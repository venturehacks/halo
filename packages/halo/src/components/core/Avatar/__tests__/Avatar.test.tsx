import { render } from 'enzyme';
import React from 'react';

import { Avatar, AvatarSize, IMAGE_SIZES } from '../index';

const sharedProps = {
  imageUrl: 'https://avatars1.githubusercontent.com/u/194885',
  name: 'Drew',
};

describe('Avatar', () => {
  test('smoke', () => {
    const component = render(<Avatar {...sharedProps} size="sm" />);

    expect(component.html()).toContain('<img');
  });

  describe('snapshots', () => {
    describe('size', () => {
      test(`xxs`, () => {
        const component = render(<Avatar {...sharedProps} size="xxs" />);

        expect(component).toMatchSnapshot();
      });

      test(`xs`, () => {
        const component = render(<Avatar {...sharedProps} size="xs" />);

        expect(component).toMatchSnapshot();
      });

      test(`md [default]`, () => {
        const component = render(<Avatar {...sharedProps} size="md" />);

        expect(component).toMatchSnapshot();
      });

      test(`lg`, () => {
        const component = render(<Avatar {...sharedProps} size="lg" />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('shape', () => {
      test(`circle [default]`, () => {
        const component = render(<Avatar {...sharedProps} size="md" />);

        expect(component).toMatchSnapshot();
      });
      test(`square`, () => {
        const component = render(
          <Avatar {...sharedProps} shape="square" size="md" />,
        );

        expect(component).toMatchSnapshot();
      });
    });

    test(`badge label`, () => {
      const component = render(
        <Avatar {...sharedProps} badge="Test Badge" size="lg" />,
      );

      expect(component.text()).toContain('Test Badge');
      expect(component).toMatchSnapshot();
    });

    test(`badgeColor`, () => {
      const component = render(
        <Avatar
          {...sharedProps}
          badge="Purple"
          badgeColor="purple"
          size="lg"
        />,
      );

      expect(component.text()).toContain('Purple');
      expect(component).toMatchSnapshot();
    });

    describe(`badgeShape`, () => {
      test(`circle [default]`, () => {
        const component = render(
          <Avatar {...sharedProps} badge="Test Badge" size="lg" />,
        );

        expect(component).toMatchSnapshot();
      });

      test(`square`, () => {
        const component = render(
          <Avatar
            {...sharedProps}
            badge="Test Badge"
            badgeShape="square"
            size="lg"
          />,
        );

        expect(component).toMatchSnapshot();
      });
    });
  });

  describe(`Dimensions`, () => {
    it(`adds the right height and width to the img tag`, () => {
      const sizes = Object.keys(IMAGE_SIZES) as AvatarSize[];

      sizes.forEach((size) => {
        const component = render(<Avatar {...sharedProps} size={size} />);
        const image = component.find('img');

        expect(image.prop('height')).toEqual(IMAGE_SIZES[size].toString());
        expect(image.prop('width')).toEqual(IMAGE_SIZES[size].toString());
      });
    });
  });
});
