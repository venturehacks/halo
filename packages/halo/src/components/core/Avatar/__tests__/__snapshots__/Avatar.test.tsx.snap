// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar snapshots badge label 1`] = `
<div
  class="component lg circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    height="72"
    src="https://avatars1.githubusercontent.com/u/194885"
    width="72"
  />
  <div
    class="badge"
  >
    <div
      class="absolute border-solid border-gray-100 text-center font-medium uppercase leading-none antialiased p-1 right-0 text-xs bg-gray-600 border-gray-100 text-white rounded-full -top-3 right-1"
    >
      <div
        class="flex justify-center gap-1"
      >
        <span>
          Test Badge
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Avatar snapshots badgeColor 1`] = `
<div
  class="component lg circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    height="72"
    src="https://avatars1.githubusercontent.com/u/194885"
    width="72"
  />
  <div
    class="badge"
  >
    <div
      class="absolute border-solid border-gray-100 text-center font-medium uppercase leading-none antialiased p-1 right-0 text-xs bg-purple-600 border-purple-100 text-white rounded-full -top-3 right-1"
    >
      <div
        class="flex justify-center gap-1"
      >
        <span>
          Purple
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Avatar snapshots badgeShape circle [default] 1`] = `
<div
  class="component lg circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    height="72"
    src="https://avatars1.githubusercontent.com/u/194885"
    width="72"
  />
  <div
    class="badge"
  >
    <div
      class="absolute border-solid border-gray-100 text-center font-medium uppercase leading-none antialiased p-1 right-0 text-xs bg-gray-600 border-gray-100 text-white rounded-full -top-3 right-1"
    >
      <div
        class="flex justify-center gap-1"
      >
        <span>
          Test Badge
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Avatar snapshots badgeShape square 1`] = `
<div
  class="component lg circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    height="72"
    src="https://avatars1.githubusercontent.com/u/194885"
    width="72"
  />
  <div
    class="badge"
  >
    <div
      class="absolute border-solid border-gray-100 text-center font-medium uppercase leading-none antialiased p-1 right-0 text-xs bg-gray-600 border-gray-100 text-white rounded-md -top-3 right-1"
    >
      <div
        class="flex justify-center gap-1"
      >
        <span>
          Test Badge
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Avatar snapshots shape circle [default] 1`] = `
<div
  class="component md circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    height="48"
    src="https://avatars1.githubusercontent.com/u/194885"
    width="48"
  />
</div>
`;

exports[`Avatar snapshots shape square 1`] = `
<div
  class="component md square"
>
  <img
    alt="Avatar for Drew"
    class="avatar square"
    height="48"
    src="https://avatars1.githubusercontent.com/u/194885"
    width="48"
  />
</div>
`;

exports[`Avatar snapshots size lg 1`] = `
<div
  class="component lg circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    height="72"
    src="https://avatars1.githubusercontent.com/u/194885"
    width="72"
  />
</div>
`;

exports[`Avatar snapshots size md [default] 1`] = `
<div
  class="component md circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    height="48"
    src="https://avatars1.githubusercontent.com/u/194885"
    width="48"
  />
</div>
`;

exports[`Avatar snapshots size xs 1`] = `
<div
  class="component xs circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    height="24"
    src="https://avatars1.githubusercontent.com/u/194885"
    width="24"
  />
</div>
`;

exports[`Avatar snapshots size xxs 1`] = `
<div
  class="component xxs circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    height="16"
    src="https://avatars1.githubusercontent.com/u/194885"
    width="16"
  />
</div>
`;
